src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.PackTool.targets

```diff
@@ -0,0 +1,56 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.PackTool.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="Microsoft.NET.Build.Tasks.GenerateToolsSettingsFile"
+              AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <!--
+    Mark all dependecy as private assets. But keep them as Publish. So dependency DLLs will be included in NuGet package, while
+    there is no dependency in nuspec. And add Microsoft.NETCore.Platforms, which is used to select correct RID assets.
+    -->
+  <ItemGroup Condition=" '$(PackAsTool)' == 'true' ">
+    <PackageReference Update="@(PackageReference)">
+      <PrivateAssets>All</PrivateAssets>
+      <Publish Condition=" '%(PackageReference.Publish)' != 'false' ">true</Publish>
+    </PackageReference>
+
+    <PackageReference Include="Microsoft.NETCore.Platforms" Version="$(NETCorePlatformsImplicitPackageVersion)" />
+  </ItemGroup>
+
+  <Target Name="PackTool" DependsOnTargets="GenerateToolsSettingsFileFromBuildProperty;Publish" Condition=" '$(PackAsTool)' == 'true' ">
+    <!--All published file in different tools/tfm-->
+    <ItemGroup>
+        <_PublishedTfmSpecificPackageFiles Include="@(ResolvedFileToPublish->'$(PublishDir)%(RelativePath)')"/>
+        <_PublishedTfmSpecificPackageFiles Include="$(PublishDepsFilePath)"/>
+        <_PublishedTfmSpecificPackageFiles Include="$(PublishRuntimeConfigFilePath)"/>
```

Comment:
> Does this cover all the files that are published?

Yes

> Are the two properties you are including not included in ResolvedFileToPublish? 

These 2 file are created for publish

> Are there other files that need to be included?

No, I checked the result with glob vs using ResolvedFileToPublish

Reply:
I think I suggested `ResolvedFileToPublish` previously but having doubts now. It could by `PreserveNewest` with some modification already done in the publish output or users could have targets with `<Copy>` stuff in there. (or even post-built steps using xcopy execute).
The most "correct" but also slowest would be to do the same as the web sdk does for publishing - clean an intermediate directory and publish to it, then use a glob to harvest this clean publish. e.g. like https://github.com/dotnet/sdk/issues/1846#issuecomment-356726385

Reply:
Let me log an issue to further discuss this. https://github.com/dotnet/sdk/issues/1853